2996_reflection/reflection.md

```diff
@@ -271,4 +183,202 @@ std::string name1 = name_of(^S);             // Also okay.
+typename[:r:] x = 42;       // Same as: int x = 42;
+typename[:^char:] c = '*';  // Same as: char c = '*';
+```
+:::
+
+The `typename` prefix can be omitted in the same contexts as with dependent qualified names.  For example:
+
+:::bq
+```c++
+using MyType = [:sizeof(int)<sizeof(long)? ^long : ^int:];  // Implicit "typename" prefix.
+```
+:::
+
+
+## Selecting Members
+
+Our second example enables selecting a member "by number" for a specific type.  It also shows the use of a metafunction dealing with diagnostics:
+
+:::bq
+```c++
+struct S { unsigned i:2, j:6; };
+
+consteval auto member_number(int n) {
+  if (n == 0) return ^S::i;
+  else if (n == 1) return ^S::j;
+  else return std::meta::invalid_reflection("Only field numbers 0 and 1 permitted");
+}
+
+int main() {
+  S s{0, 0};
+  s.[:member_number(1):] = 42;  // Same as: s.j = 42;
+  s.[:member_number(5):] = 0;   // Error (likely with "Only field numbers 0 and 1 permitted" in text).
+}
+```
+:::
+
+This example also illustrates that bit fields are not beyond the reach of this proposal.
+
+
+## Fast Generation of Integer Sequence
+
+:::bq
+```c++
+#include <utility>
+#include <vector>
+
+template<typename T>
+consteval info make_integer_seq_refl(T N) {
+  std::vector args{^T};
+  for (T k = 0; k<N; ++k)  args.push_back(std::meta::reflect_value(k));
+  return substitute(^std::integer_sequence, args);
+}
+
+template<typename T>
+  using make_integer_sequence<typename T, T N> = [:make_integer_seq_refl<T>(N):];
+```
+:::
+
+
+## Enum to String
+
+One of the most commonly requested facilities is to convert an enum value to a string (this example relies on expansion statements [@P1306R1]):
+
+::: bq
+```c++
+template <typename E>
+  requires std::is_enum_v<E>
+constexpr std::string enum_to_string(E value) {
+  template for (constexpr auto e : std::meta::members_of(^E)) {
+    if (value == [:e:]) {
+      return std::string(std::meta::name_of(e));
+    }
+  }
+
+  return "<unnamed>";
+}
+
+enum Color { red, green, blue };
+static_assert(enum_to_string(Color::red) == "red");
+static_assert(enum_to_string(Color(42)) == "<unnamed>");
+```
+:::
+
+We can also do the reverse in pretty much the same way:
+
+::: bq
+```c++
+template <typename E>
+  requires std::is_enum_v<E>
+constexpr std::optional<E> string_to_enum(std::string_view name) {
+  template for (constexpr auto e : std::meta::members_of(^E)) {
+    if (name == std::meta::name_of(e)) {
+      return [:e:];
+    }
+  }
+
+  return std::nullopt;
+}
+```
+:::
+
+
+## Parsing Command-Line Options
+
+Our next example shows how command-line options could be automatically mapped to an "options" structure.
+For simplicity, we posit the existence of a range-like class type `ProgramArgs` that collects the traditional `(argc, argv)` parameters in a more friendly package.
+This example also uses an expansion statement for simplicity.
+
+::: bq
+```c++
+#include <sstream>
+#include <string>
+#include <meta>
+#include <ProgramArgs.h>
```

Comment:
I would like to avoid examples that include random headers that readers won't know about. What's `ProgramArgs`? I think it's a lot better here to just use `std::vector<std::string_view>` here.

Also I think the right signature for this is:
```cpp
template <typename Opts>
auto parse_options(int argc, char** argv) -> Opts;
```

Just `std::exit` on failure. That's probably what you want anyway and it makes for a simpler example here.